src/init.cpp

```diff
@@ -1924,6 +1924,8 @@ bool AppInitMain(const CoreContext& context, NodeContext& node, interfaces::Bloc
                 node.evodb = std::make_unique<CEvoDB>(nEvoDbCache, false, fReset || fReindexChainState);
                 node.mnhf_manager.reset();
                 node.mnhf_manager = std::make_unique<CMNHFManager>(*node.evodb);
+                node.creditPoolManager.reset();
+                node.creditPoolManager = std::make_unique<CCreditPoolManager>(*node.evodb);
```

Comment:
Why is this here? Is it an artifact of a rebase or something?

Reply:
that's indeed duplicate of re-init, see https://github.com/dashpay/dash/pull/5776/files#diff-b1e19192258d83199d8adaa5ac31f067af98f63554bfdd679bd8e8073815e69dL1943

```suggestion
```

Reply:
No, this doesn't work. RPC needs to access to `node.creditPoolManager`

Reply:
indeed. Good catch! Though, 2 objects should not co-exist in this cruel world.
here's hot fix: https://github.com/knst/dash/commit/648ac85cea4b5f15ebc4be8ff20b37b90fc02ee0
Proper fix requires deglobalization of `creditPoolManager`, out-of-score for this PR